### DISCUSSION TITLE
How do disable shader compiler warnings

### DISCUSSION THREAD

**Discussion by molysgaard** [google/filament#5372] [Q&A] [answered]
Hi, we are working on updating the version of filament we are using in our project.

During this update, there is some new compiler warnings are output from the filament shader compiler during debug builds:

```
WARNING: :0:0:0: The function '%263 = OpFunction %6 None %257' could not be inlined because the return instruction is not at the end of the function. This could be fixed by running merge-return before inlining.
WARNING: :0:0:0: The function '%435 = OpFunction %23 None %68' could not be inlined because the return instruction is not at the end of the function. This could be fixed by running merge-return before inlining.
WARNING: :0:0:0: The function '%435 = OpFunction %23 None %69' could not be inlined because the return instruction is not at the end of the function. This could be fixed by running merge-return before inlining.
```

These outputs come from calling the MaterialBuilder API like this:

```cpp
    filamat::Package defaultShader = MaterialBuilderHelper()
                .materialFromFile("./shaders/Default.frag")
                .name("DefaultShader")
                .optimization(filamat::MaterialBuilderBase::Optimization::PERFORMANCE)
                .generateDebugInfo(false)
                .require(filament::VertexAttribute::POSITION)
                .blending(filament::BlendingMode::MASKED)
                .parameter(filamat::MaterialBuilder::UniformType::FLOAT4, "RGBA")
                .parameter(filamat::MaterialBuilder::UniformType::FLOAT, "clearCoat")
                .parameter(filamat::MaterialBuilder::UniformType::FLOAT, "roughness")
                .shading(filament::Shading::LIT).build(_engine.getJobSystem());
```

The warnings are not output if we compile our application in `Release` mode.

*Is there any setting to disable the warning output?*

Currently the amount of printed text is so large that in overflows any useful output in the terminal.

**Comment by bejado** [comment, marked as answer]
Unfortunately we don't have a setting for this, but you can modify [this function](https://github.com/google/filament/blob/d879bbc80a30ad95d8439057198cf44a48d4012e/libs/filamat/src/GLSLPostProcessor.cpp#L111-L121) and re-compile Filament.

**Reply by molysgaard** [reply]
Thanks for the answer. We ended up changing our build pipeline to pre-build our shaders compile time using `matc`. It had other added benefits of reducing startup time of our application in addition to getting rid of the debug output while our application runs.

I can imagine applications that generate and compile shader code live while they are running would not be able to do this though. They would still have quite a console-spam problem in `Debug` mode.
Anyways, the problem is solved for our use-case. Thanks!
